crypto/asn1/a_time.c

```diff
@@ -489,7 +489,7 @@ int ASN1_TIME_print(BIO *bp, const ASN1_TIME *tm)
         if (tm->length > 15 && v[14] == '.') {
             f = &v[14];
             f_len = 1;
-            while (14 + f_len < l && isdigit(f[f_len]))
+            while (14 + f_len < l && ossl_isdigit(f[f_len]))
```

Comment:
If EBCDIC is enabled, the ossl_isXXX functions convert the character to ASCII using `os_toascii` before checking for character class membership.  This should work for the two different EBCDIC tables and the dummy test table.